### DISCUSSION TITLE
Initalize model data by values set in plain html

### DISCUSSION THREAD

**Discussion by MartinMystikJonas** [alpinejs/alpine#2049] [1. Help] [answered]
What is the best practice to initalize model data by values that are available in plain html before Apline takes over.

For example how to set model value to input value or html attribute value?

It seems inut values and htmla ttributes are always overwritten by default model values set in x-data before I can access them (for example by x-init). 

In out use case it is not easy to know all model attributes at the moment we render x-data (it is complex component rendered across different templates) but all required values are in rendered html.

We yould need something to initalize model data by values we would read from html before Apline overwrites them with default model values.

Something like ```<input value="foo" x-model="value" x-before-init="value=$el.value" />``` would be best.

Can somebody please show me how this is solved? Thanks.

**Comment by SimoTod** [comment, marked as answer]
X-init just works for your use case: https://codepen.io/SimoTod/pen/WNOjdZo

**Reply by MartinMystikJonas** [reply]
Dammit it didn't work in my code . But I guess error is somewhere else. Thanks.
